gslib/tests/test_stet_cp.py

```diff
@@ -0,0 +1,146 @@
+
+    stdout = self.RunGsUtil(['cat', suri(object_uri)], return_stdout=True)
+    self.assertIn('subcommand: encrypt', stdout)
+    self.assertIn('config file: --config-file={}'.format(self.stet_config_path),
+                  stdout)
+    self.assertIn('blob id: --blob-id={}'.format(suri(object_uri)), stdout)
+    self.assertIn('in file: {}'.format(test_file), stdout)
+    self.assertIn('out file: {}_.stet_tmp'.format(test_file), stdout)
+
+    self.assertFalse(
+        os.path.exists(
+            temporary_file_util.GetStetTempFileName(
+                storage_url.StorageUrlFromString(test_file))))
+
+  def test_decrypts_download_if_stet_is_enabled(self):
+    object_uri = self.CreateObject(contents='abc')
+    test_file = self.CreateTempFile()
+
+    stderr = self.RunGsUtil([
+        '-o', 'GSUtil:stet_binary_path={}'.format(self.stet_binary_path), '-o',
+        'GSUtil:stet_config_path={}'.format(
+            self.stet_config_path), 'cp', '--stet',
+        suri(object_uri), test_file
+    ],
+                            return_stderr=True)
+
+    # Progress indicator should show transformed file size (variable based on
+    # random string generation above). 4.0 B is the pre-transform size.
+    self.assertNotIn('/4.0 B]', stderr)
+
+    with open(test_file) as file_reader:
+      downloaded_text = file_reader.read()
+    self.assertIn('subcommand: decrypt', downloaded_text)
+    self.assertIn('config file: --config-file={}'.format(self.stet_config_path),
+                  downloaded_text)
+    self.assertIn('blob id: --blob-id={}'.format(suri(object_uri)),
+                  downloaded_text)
+    self.assertIn('in file: {}'.format(test_file), downloaded_text)
+    self.assertIn('out file: {}_.stet_tmp'.format(test_file), downloaded_text)
+
+    self.assertFalse(
+        os.path.exists(
+            temporary_file_util.GetStetTempFileName(
+                storage_url.StorageUrlFromString(test_file))))
+
+  def test_does_not_seek_ahead_for_bytes_if_stet_transform(self):
+    """Tests that cp does not seek-ahead for bytes if file size will change."""
+    tmpdir = self.CreateTempDir()
+    for _ in range(3):
+      self.CreateTempFile(tmpdir=tmpdir, contents=b'123456')
+
+    bucket_uri = self.CreateBucket()
+
+    with util.SetBotoConfigForTest([
+        ('GSUtil', 'task_estimation_threshold', '1'),
+        ('GSUtil', 'task_estimation_force', 'True')
+    ]):
+      stderr = self.RunGsUtil([
+          '-m', '-o', 'GSUtil:stet_binary_path={}'.format(
+              self.stet_binary_path), '-o', 'GSUtil:stet_config_path={}'.format(
+                  self.stet_config_path), 'cp', '-r', '--stet', tmpdir,
+          suri(bucket_uri)
+      ],
+                              return_stderr=True)
+      # Pre-encryption total file size that seek-ahead would find.
+      self.assertNotIn('18.0 B]', stderr)
+      # Post-encryption total file size found if seek-ahead is disabled.
+      self.assertIn('2.0 KiB]', stderr)
```

Comment:
Sry, I still don't understand how this test works. This is showing the updated size, but how does this prove that the seek ahead did not calculate the size? I have a feeling that you will get the same output even if you explicitly set `self.self.count_data_bytes` to True?

Also, full text for the progress callback output might be helpful to read what exactly this represents. something like
```
[3 files][   32.0 B/   32.0 B]
```

Reply:
Added more comments

Reply:
Talked to Nick offline and he clarified how this is intended to work. Looks good.